js/ui/codingManager.js

```diff
@@ -210,20 +214,64 @@ const WindowTrackingButton = new Lang.Class({
     // Just fade out and fade the button back in again. This makes it
     // look as though we have two buttons, but in reality we just have
     // one.
-    switchAnimation: function() {
+    switchAnimation: function(direction) {
+        let rect = this.window.get_frame_rect();
+
+        // this API is deprecated but the best option here to
+        // animate around a point outside of the actor
+        this.rotation_center_y = new Clutter.Vertex({ x: this.width - (rect.width * 0.5),
+                                                      y: this.height * 0.5,
+                                                      z: 0 });
+        this.rotation_angle_y = 0;
+        Tweener.addTween(this, {
+            rotation_angle_y: direction == Gtk.DirectionType.RIGHT ? 180 : -180,
+            time: WINDOW_ANIMATION_TIME * 4,
+            transition: 'easeOutQuad'
+        });
         Tweener.addTween(this, {
             opacity: 0,
-            time: WINDOW_ANIMATION_TIME / 2,
+            time: WINDOW_ANIMATION_TIME * 2,
             transition: 'linear',
             onComplete: Lang.bind(this, function() {
+                Tweener.removeTweens(this);
+                this.rotation_angle_y = 0;
                 Tweener.addTween(this, {
                     opacity: 255,
-                    time: WINDOW_ANIMATION_TIME / 2,
+                    time: WINDOW_ANIMATION_TIME * 0.5,
                     transition: 'linear',
-                    delay: WINDOW_ANIMATION_TIME * 3
+                    delay: WINDOW_ANIMATION_TIME * 1.5
                 });
             })
         });
+
+        let button = new St.Bin({ style_class: 'view-source',
+                                  x_fill: true,
+                                  y_fill: false });
+        button.set_child(_createIcon());
+
+        Main.layoutManager.addChrome(button);
+        button.set_position(rect.x + rect.width - BUTTON_OFFSET_X,
+                            rect.y + rect.height - BUTTON_OFFSET_Y);
```

Comment:
Did you see my comment earlier about sharing this code with the code that creates the original button?